src/ir/sc.rs

```diff
@@ -63,10 +75,30 @@ impl<F: Field> MappingContext<F> {
 pub type Mapping<F, MappingArgs> = dyn Fn(&mut MappingContext<F>, MappingArgs) + 'static;
 
 pub struct MappingGenerator<F, MappingArgs> {
-    mapping: Mapping<F, MappingArgs>,
+    mapping: Rc<Mapping<F, MappingArgs>>,
+}
+
+impl<F, MappingArgs> Clone for MappingGenerator<F, MappingArgs> {
+    fn clone(&self) -> Self {
+        Self {
+            mapping: self.mapping.clone(),
+        }
+    }
+}
```

Comment:
Can be derived?

Reply:
For deriving it, it requires F and MappingArgs to be Clone, which is not really necessary.